### ISSUE TITLE
fermata on chord not working

### ISSUE THREAD

**Issue by rettinghaus** [rism-ch/verovio#348]
`@fermata` on `<chord>` doesn't get drawn. If it's only present on a `<note>` within `<chord>` (which *is* bad MEI style, I know) it can collide with other notes.

**Comment by craigsapp** [comment]
Demonstration of the problem:

<img width="156" alt="screen shot 2016-11-11 at 08 20 10" src="https://cloud.githubusercontent.com/assets/3487289/20221824/beba5c62-a7e7-11e6-8d2c-b829c8972af4.png">

Generated with:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="http://music-encoding.org/schema/3.0.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="http://music-encoding.org/schema/3.0.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="3.0.0">
    <meiHead>
        <fileDesc>
            <titleStmt>
                <title />
            </titleStmt>
            <pubStmt>
                <date>2016-11-11 08:19:45</date>
            </pubStmt>
        </fileDesc>
        <encodingDesc>
            <projectDesc>
                <p>Transcoded from Humdrum with Verovio version 0.9.13-dev-08c6405-dirty</p>
            </projectDesc>
        </encodingDesc>
    </meiHead>
    <music>
        <body>
            <mdiv>
                <score>
                    <scoreDef xml:id="scoredef-0000001925940357">
                        <staffGrp xml:id="m-0000001001221272">
                            <staffDef xml:id="staffdef-0000002136289730" clef.shape="G" clef.line="2" n="1" lines="5" />
                        </staffGrp>
                    </scoreDef>
                    <section xml:id="section-0000001250015842">
                        <measure xml:id="measure-L1" n="0">
                            <staff xml:id="staff-0000002138135129" n="1">
                                <layer xml:id="layer-L1F1" n="1">
                                    <chord xml:id="chord-L3F1" dur="1">
                                        <note xml:id="note-L3F1S1" fermata="above" oct="5" pname="a" accid.ges="n" />
                                        <note xml:id="note-L3F1S2" fermata="above" oct="4" pname="a" accid.ges="n" />
                                    </chord>
                                </layer>
                            </staff>
                        </measure>
                    </section>
                </score>
            </mdiv>
        </body>
    </music>
</mei>
```

Trying to move the fermata to the chord:

<img width="168" alt="screen shot 2016-11-11 at 08 22 32" src="https://cloud.githubusercontent.com/assets/3487289/20221888/02a826de-a7e8-11e6-8d0e-a9fd5c6150b0.png">


```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="http://music-encoding.org/schema/3.0.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="http://music-encoding.org/schema/3.0.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="3.0.0">
    <meiHead>
        <fileDesc>
            <titleStmt>
                <title />
            </titleStmt>
            <pubStmt>
                <date>2016-11-11 08:19:45</date>
            </pubStmt>
        </fileDesc>
        <encodingDesc>
            <projectDesc>
                <p>Transcoded from Humdrum with Verovio version 0.9.13-dev-08c6405-dirty</p>
            </projectDesc>
        </encodingDesc>
    </meiHead>
    <music>
        <body>
            <mdiv>
                <score>
                    <scoreDef xml:id="scoredef-0000001925940357">
                        <staffGrp xml:id="m-0000001001221272">
                            <staffDef xml:id="staffdef-0000002136289730" clef.shape="G" clef.line="2" n="1" lines="5" />
                        </staffGrp>
                    </scoreDef>
                    <section xml:id="section-0000001250015842">
                        <measure xml:id="measure-L1" n="0">
                            <staff xml:id="staff-0000002138135129" n="1">
                                <layer xml:id="layer-L1F1" n="1">
                                    <chord xml:id="chord-L3F1" fermata="above" dur="1">
                                        <note xml:id="note-L3F1S1" oct="5" pname="a" accid.ges="n" />
                                        <note xml:id="note-L3F1S2" oct="4" pname="a" accid.ges="n" />
                                    </chord>
                                </layer>
                            </staff>
                        </measure>
                    </section>
                </score>
            </mdiv>
        </body>
    </music>
</mei>
```

Trying `<fermata>` as a child of `<chord>`:

<img width="168" alt="screen shot 2016-11-11 at 08 22 32" src="https://cloud.githubusercontent.com/assets/3487289/20221927/377e9672-a7e8-11e6-9575-82ef622a8840.png">

```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="http://music-encoding.org/schema/3.0.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="http://music-encoding.org/schema/3.0.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="3.0.0">
    <meiHead>
        <fileDesc>
            <titleStmt>
                <title />
            </titleStmt>
            <pubStmt>
                <date>2016-11-11 08:19:45</date>
            </pubStmt>
        </fileDesc>
        <encodingDesc>
            <projectDesc>
                <p>Transcoded from Humdrum with Verovio version 0.9.13-dev-08c6405-dirty</p>
            </projectDesc>
        </encodingDesc>
    </meiHead>
    <music>
        <body>
            <mdiv>
                <score>
                    <scoreDef xml:id="scoredef-0000001925940357">
                        <staffGrp xml:id="m-0000001001221272">
                            <staffDef xml:id="staffdef-0000002136289730" clef.shape="G" clef.line="2" n="1" lines="5" />
                        </staffGrp>
                    </scoreDef>
                    <section xml:id="section-0000001250015842">
                        <measure xml:id="measure-L1" n="0">
                            <staff xml:id="staff-0000002138135129" n="1">
                                <layer xml:id="layer-L1F1" n="1">
                                    <chord xml:id="chord-L3F1" dur="1">
                                        <fermata/>
                                        <note xml:id="note-L3F1S1" oct="5" pname="a" accid.ges="n" />
                                        <note xml:id="note-L3F1S2" oct="4" pname="a" accid.ges="n" />
                                    </chord>
                                </layer>
                            </staff>
                        </measure>
                    </section>
                </score>
            </mdiv>
        </body>
    </music>
</mei>
```

`<fermata>` is not allowed as a child element of `<note>` or `<chord>`?

http://music-encoding.org/documentation/3.0.0/fermata

**Comment by pe-ro** [comment]
`<fermata>` is a particular kind of "directive" (like `<tempo>`, `<harm>`, or `<ornam>`) that I call "control events".  As such, members of this class sit outside the stream of events (notes and rests) and aren't allowed as children of events.

**Comment by craigsapp** [comment]
How should a fermata be applied to a chord?

**Comment by rettinghaus** [comment]
With tstamp or startid.

**Comment by craigsapp** [comment]
A more complete example within a chord would be more useful.

**Comment by rettinghaus** [comment]
Yeah, sorry! Something like this: 

```xml
    <music>
        <body>
            <mdiv>
                <score>
                    <scoreDef xml:id="scoredef-0000001925940357">
                        <staffGrp xml:id="m-0000001001221272">
                            <staffDef xml:id="staffdef-0000002136289730" clef.shape="G" clef.line="2" n="1" lines="5" />
                        </staffGrp>
                    </scoreDef>
                    <section xml:id="section-0000001250015842">
                        <measure xml:id="measure-L1" n="0">
                            <staff xml:id="staff-0000002138135129" n="1">
                                <layer xml:id="layer-L1F1" n="1">
                                    <chord xml:id="chord-L3F1" dur="1">
                                        <note xml:id="note-L3F1S1" oct="5" pname="a" accid.ges="n" />
                                        <note xml:id="note-L3F1S2" oct="4" pname="a" accid.ges="n" />
                                    </chord>
                                </layer>
                            </staff>
                            <fermata place="above" startid="chord-L3F1"/>
                        </measure>
                    </section>
                </score>
            </mdiv>
        </body>
    </music>
```

**Comment by craigsapp** [comment]
Thanks!  Although I will complain to @pe-ro that that seems a bit cumbersome for the case of applying a fermata to a chord.  

Here is a case where a fermata directive is misleading:

<img width="357" alt="screen shot 2016-11-11 at 11 27 45" src="https://cloud.githubusercontent.com/assets/3487289/20227636/fd274e3c-a801-11e6-9c2e-e22d0ef8fd5b.png">

Using a directive for the first fermata is somewhat implying that it applies to all voices/layers on the staff, when it only applies to the half-note chord and not to the quarter note occurring at the same time.

`chord@fermata` is allowed, so that is probably how I would do the above case (unless further attributes on the fermata are needed).

It seems that having a fermata as a child element of chord/note/rest is more logical when extra parameters for the fermata is needed, rather than promoting the fermata to a directive.

Here is a case where a directive fermata is the most appropriate:

<img width="177" alt="screen shot 2016-11-11 at 11 41 58" src="https://cloud.githubusercontent.com/assets/3487289/20227996/055639a4-a804-11e6-8ea1-79409fbd4f4d.png">

In this case the inner voices do not have fermatas due to space limitations, and the fermata on the outer voices are also implied to be applied to the inner voice notes when expressed as a directive.

**Comment by rettinghaus** [comment]
I see your point, @craigsapp ! But for the latter you would use `@plist` to show that the fermata applies to two notes. So if you want to show that the fermata is *only* meant for the chord, you should use `@plist`, too.

**Comment by craigsapp** [comment]
`@plist` will work nicely for the examples I gave.  That is probably better to use on fermata directives rather than `@startid`, since they typically would not have `@endid`s (at least I cannot imagine how they would).

**Comment by lpugin** [comment]
Fixed in ed3f8e838de9ffcc37ce5f57554fe044c81c7e80

**Comment by ahankinson** [comment]
I'll also point out that `@staff` and `@layer` are available on fermata for further clarifying its relationships
